go-controller/pkg/metrics/ovn.go

```diff
@@ -342,6 +349,65 @@ func getPortCount(portType string) float64 {
 	return portCount
 }
 
+//ovnControllerSBDBConnectionCheckUpdater blocks until stopCh closed but before then polls ovn-controllers connection status with
+//southbound database periodically.
+func ovnControllerSBDBConnectionCheckUpdater(stopCh <-chan struct{}, ovsAppctl ovsClient, period time.Duration) {
+	// There maybe transient connection issues to SB DB. We want to minimise the risk of reporting this as the current state between
+	// long poll intervals.
+	retry := 5
+	sleepBetweenRetry := 5 * time.Second
+	// retry times sleepBetweenRetry
+	totalRetriesTime := 25 * time.Second
```

Comment:
don't see any real usage of this var

Reply:
As you saw, its used to check if the period func arg is less than the retry time, and panic if its the case.

I am happy to hear you'd rework it.

Reply:
ah lol
what about `totalRetriesTime := retry * sleepBetweenRetry`

Reply:
+1

Reply:
you cannot multiple ints and time.Duration but Ill rework this because you both dont seem to like it.

Reply:
I added a beautiful cast :) Don't know if I prefer this over what I had originally..

Reply:
`retryTotal := retrySleep * time.Duration(retry)`